ansible/roles/operator-pipeline/tasks/main.yml

```diff
@@ -49,25 +50,29 @@
         operator-pipeline.key: "{{ lookup('file', operator_pipeline_private_key_local_path, rstrip=False) | b64encode }}"
         operator-pipeline.pem: "{{ lookup('file', operator_pipeline_private_cert_local_path, rstrip=False) | b64encode }}"
 
-
-- name: Create Operator pipeline quay token secret
-  no_log: yes
-  k8s:
-    state: present
-    force: yes
-    namespace: "{{ oc_namespace }}"
-    definition:
-      apiVersion: v1
-      kind: Secret
-      type: kubernetes.io/dockerconfigjson
-      metadata:
-        name: operator-pipeline-registry-token-{{ suffix }}
-        labels:
-          app: operator-pipeline
-          suffix: "{{ suffix }}"
-          env: "{{ env }}"
-      data:
-        .dockerconfigjson: "{{ lookup('file', operator_pipeline_quay_token, rstrip=False) | b64encode }}"
+- name: Grant registry access
+  block:
+    - name: Create Operator pipeline quay token secret
+      no_log: yes
+      k8s:
+        state: present
+        force: yes
+        namespace: "{{ oc_namespace }}"
+        definition:
+          apiVersion: v1
+          kind: Secret
+          type: kubernetes.io/dockerconfigjson
+          metadata:
+            name: operator-pipeline-registry-token-{{ suffix }}
+            labels:
+              app: operator-pipeline
+              suffix: "{{ suffix }}"
+              env: "{{ env }}"
+          data:
+            .dockerconfigjson: "{{ lookup('file', operator_pipeline_quay_token, rstrip=False) | b64encode }}"
+    - name: Link secret with pipeline service account (for imagestream operations)
+      ansible.builtin.shell: |
+        oc secret link pipeline registry-dockerconfig-secret
```

Comment:
How is this SA created? Does it come with a pipeline operator?

Reply:
yes, 
```
The OpenShift Pipelines Operator adds and configures a Serviceaccount named pipeline that has sufficient permissions to build and push an image. This ServiceAccount is used by pipelineruns.
```
https://openshift.github.io/pipelines-docs/docs/0.10.5/assembly_creating-applications-with-cicd-pipelines.html